CMakeLists.txt

```diff
@@ -1,52 +1,19 @@
-cmake_minimum_required(VERSION 3.11)
+cmake_minimum_required(VERSION 3.28)
 project(gpu)
 
-include(FetchContent)
-message(STATUS "CMAKE_CURRENT_SOURCE_DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
-set(FETCHCONTENT_BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third_party/fetchcontent")
+include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/webgpu.cmake")
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON) # export compile_commands.json to use with LSP
-set(CMAKE_CXX_STANDARD 17)
```

Comment:
This might be workable but there's potentially issues with the out-of-the-box clang compiler on mac not supporting some c++20 features. If it's possible to stay on 17 and still build things that would be preferable, i can take a closer look tomorrow.

Reply:
We should be able to get by with 17, will do a test build and commit.